tests/aws/services/events/conftest.py

```diff
@@ -566,7 +566,12 @@ def _create_connection(auth_type_or_auth, auth_parameters=None):
                 AuthParameters=auth_parameters,
             )
 
-    return _create_connection
+    yield _create_connection
+
+    try:
+        aws_client.events.delete_connection(Name=connection_name)
+    except Exception as e:
+        LOG.debug("Error cleaning up connection: %s", e)
```

Comment:
Nice catch. Kinda reminding me of the fact that it  might be nice to revive the client-based cleanups at some point :grin: